app/src/androidTest/java/ch/epfl/skysync/screens/reports/CrewReportScreenTest.kt

```diff
@@ -1,29 +1,118 @@
 package ch.epfl.skysync.screens.reports
 
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.test.hasTestTag
 import androidx.compose.ui.test.junit4.createComposeRule
-import ch.epfl.skysync.models.calendar.AvailabilityCalendar
-import ch.epfl.skysync.models.calendar.FlightGroupCalendar
-import ch.epfl.skysync.models.flight.RoleType
-import ch.epfl.skysync.models.reports.CrewReport
-import ch.epfl.skysync.models.user.Crew
-import java.sql.Time
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollToNode
+import androidx.compose.ui.test.performTextClearance
+import androidx.compose.ui.test.performTextInput
+import androidx.navigation.compose.ComposeNavigator
+import androidx.navigation.compose.NavHost
+import androidx.navigation.testing.TestNavHostController
+import ch.epfl.skysync.Repository
+import ch.epfl.skysync.database.DatabaseSetup
+import ch.epfl.skysync.database.FirestoreDatabase
+import ch.epfl.skysync.navigation.Route
+import ch.epfl.skysync.navigation.homeGraph
+import ch.epfl.skysync.utils.inputTimePicker
+import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 
 class CrewReportScreenTest {
+
   @get:Rule val composeTestRule = createComposeRule()
+  private lateinit var navController: TestNavHostController
+  private val db = FirestoreDatabase(useEmulator = true)
+  private val dbs = DatabaseSetup()
+  private val repository = Repository(db)
+
+  @Before
+  fun setUp() {
+    composeTestRule.setContent {
+      navController = TestNavHostController(LocalContext.current)
+      navController.navigatorProvider.addNavigator(ComposeNavigator())
+      NavHost(navController = navController, startDestination = Route.MAIN) {
+        homeGraph(repository, navController, dbs.crew1.id)
+      }
+      navController.navigate(Route.CREW_REPORT)
+    }
+  }
+
+  @Test
+  fun allFieldsAreDisplayed() {
+    composeTestRule
+        .onNodeWithTag("Crew Report LazyColumn")
+        .performScrollToNode(hasTestTag("Number of little champagne bottles"))
+    composeTestRule.onNodeWithTag("Number of little champagne bottles").assertExists()
```

Comment:
Is it really useful to check if the node exists if it managed to scroll to it ?

Reply:
Not really, but, in my opinion, it makes clearer what the test is actually testing on this component